docs/kmigrator.md

```diff
@@ -0,0 +1,168 @@
+# kmigrator
+
+It's CLI migration tool. It helps you to create a `migrations` folder.
+
+Kmigrater also allow you to manage database level options. 
+You can change a table and column options. For tables, you can set constraints and indexes.
+
+## kmigrator columns options
+
+you can set options: 
+ - `null` -- to allow for this column to be nullable
+ - `unique` -- create db level unique constraint for this column
+ - `db_index` -- create db level index for this column
+ - `on_delete` -- for foreign key on delete behaviour (see details below) 
+
+Example:
+
+```
+const User = new GQLListSchema('User', {
+    schemaDoc: 'Individual / person / service account / impersonal company account',
+    fields: {
+        ...
+        email: {
+            schemaDoc: 'Email. Transformed to lower case',
+            type: Text,
+            kmigratorOptions: { null: false, unique: true },
+        },
+        birth: {
+            type: DateTimeUtc,
+            kmigratorOptions: { db_index: true },
+        },
+        ...
+    },
+})
+```
+
+#### on_delete
+
+It set relations on delete behavior.
+
+For example, if you have foregin key from Ticket.assignee to User.id you need to set 
+on delete User behaviour for Ticket.assignee column. 
+ 
+You can use options:
+
+ - `models.PROTECT` -- forbid the deletion of the referenced object. To delete it you will have to delete all objects that reference it manually. SQL equivalent: `RESTRICT`.
+ - `models.SET_NULL` -- set the reference to NULL (requires the field to be nullable). For instance, when you delete a User, you might want to keep the comments he posted on blog posts, but say it was posted by an anonymous (or deleted) user. SQL equivalent: SET NULL.
+ - `models.CASCADE` -- when the referenced object is deleted, also delete the objects that have references to it (when you remove a blog post for instance, you might want to delete comments as well). SQL equivalent: CASCADE.
+
+## kmigrator table options
+
+you can set options: 
+ - `constraints` -- to create database level constrains
+ - `indexes` -- to create database level indexes
+
+#### constraints
+
+Example:
+
+```
+const User = new GQLListSchema('User', {
+    schemaDoc: 'Individual / person who have some access to service',
+    fields: {
+        type: {
+            schemaDoc: 'We have b2b and b2c user segments',
+            type: Select,
+            dataType: 'enum',
+            options: ['b2b', 'b2c', 'sys'],
+            defaultValue: 'b2b',
+            isRequired: true,
+        },
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            isRequired: true,
+        },
+        ...
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
```

Comment:
Can we simplify interface for developer and not expose internal kmigrator conventions?
```suggestion
                type: 'unique',
```

Reply:
I don't think that it's a good idea. Because it uses `models.UniqueConstraint` under the hood. 

It's better to do it by another task. We also need to move from the `django` `Q` object syntax too in this task.